### DISCUSSION TITLE
Updating snowstorm

### DISCUSSION THREAD

**Discussion by charley-earlytracks** [IHTSDO/snowstorm#622] [Q&A] [answered]
Hello,

We are using currently Snowstorm 7.6.0.
We did not use the docker image but the plain installation.
I was wondering if there was a good practice to update the software.
Should I incrementally switch from release to release?

Thanks for your input and the good work!
Charley

**Comment by kaicode** [comment, marked as answer]
Snowstorm uses semantic version numbers with the format X.Y.Z. When the first number (X) changes there are probably breaking changes. Any breaking changes will be detailed in the release notes together with any steps required to upgrade. For example a new version of Java or Elasticsearch is required.

If you have an older version of Snowstorm I recommend upgrading incrementally using each major release. There is no need to take every minor or patch release (when only the Y or Z number changes).

So to upgrade from 7.6.0 to version 10.4.2 (latest at time of writing) take versions 8.0.0, 9.2.0, 10.1.0, and finally 10.4.2. For each upgrade make sure to read the breaking changes and upgrade requirements in the [release notes](https://github.com/IHTSDO/snowstorm/releases).
